bfvmm/include/hve/arch/intel_x64/delegator/unhandled_exit.h

```diff
@@ -0,0 +1,72 @@
+//
+// Copyright (C) 2019 Assured Information Security, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+#ifndef BFVMM_DELEGATOR_UNHANDLED_INTEL_X64_H
+#define BFVMM_DELEGATOR_UNHANDLED_INTEL_X64_H
+
+#include <stdint.h>
+#include <bfdelegate.h>
+#include <bfgsl.h>
+#include <bftypes.h>
+
+namespace bfvmm::intel_x64::unhandled_exit
+{
+
+/// Unhandled exit delegator
+///
+/// Delegates processing of vmexits that were not handled by any exit handlers
+///
+class delegator
+{
+
+public:
+
+    /// Default Constructor
+    ///
+    /// @expects none
+    /// @ensures none
+    ///
+    delegator() = default;
+
+    /// Destructor
+    ///
+    /// @expects none
+    /// @ensures none
+    ///
+    ~delegator() = default;
+
+    /// Handle
+    ///
+    /// Handle an unhandled vmexit using registered handlers
```

Comment:
Can we use a different name for this? Like 'default' or 'fallback'?

Reply:
Sure, I like fallback